nncf/torch/quantization/weights_compression.py

```diff
@@ -0,0 +1,90 @@
+from typing import Dict, Optional
+
+import torch
+from torch import nn
+
+from nncf.torch.layers import NNCF_WRAPPED_USER_MODULES_DICT
+from nncf.torch.layers import NNCFEmbedding
+from nncf.torch.layers import NNCFLinear
+from nncf.torch.quantization.quantize_functions import get_scale_zp_from_input_low_input_high
+
+
+class WeightsCompressor(nn.Module):
+    """Class provided fake quantize operation or dequantization of compressed weights in forward pass
+
+    Atributes:
+        zero_point: zero point in quantization scheme
+        scale: scale in quantizatin scheme
+    """
+
+    def __init__(self, zero_point, scale):
+        super().__init__()
+        self.zero_point = zero_point
+        self.scale = scale
+
+    def forward(self, input, op_arg):
+        if input.weight.dtype is torch.uint8:
+            w = input.weight.type(dtype=self.scale.dtype)
+            input.weight = (w - self.zero_point) * self.scale
```

Comment:
Is `input` actually layer? It looks confusing when weights are taken from input, usually weights are multiplied with input. IMO, `layer` or `module` is a better name. What do you think?

Reply:
Done

Reply:
Done